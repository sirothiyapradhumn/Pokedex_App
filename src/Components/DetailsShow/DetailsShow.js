import React from 'react'
import {Link} from 'react-router-dom'

function DetailsShow({pokeData}) {
    //console.log(pokeData);
  return (
    <div className='detailShow'>
        <div className="details-card" >
            <img src={pokeData.img} alt={pokeData.name}/>
            <div className="details-container">
                <h4><b>ID: {pokeData.num}</b></h4>
                <p className='poke-name'><b>{pokeData.name}</b></p> 
                <p className='poke-type'>Types: {pokeData.type.map((typ)=>typ+"  ")}</p>
                <p className='poke-weak'>Weaknesses: {pokeData.weaknesses.map((weak)=>weak+"  ")}</p>
                <p className='poke-hw'>Height: {pokeData.height}</p>
                <p className='poke-hw'>Weight: {pokeData.weight}</p>
                <p className='poke-np'>
                  Next/Prev Evolution: {pokeData.hasOwnProperty('next_evolution') ? pokeData.next_evolution[0].name  : pokeData.prev_evolution[0].name}
                  </p>
            </div>
        </div>

        <button><Link to="/" style={{textDecoration:"none"}}>Back</Link></button>
    </div>
    
  )
}

export default DetailsShow