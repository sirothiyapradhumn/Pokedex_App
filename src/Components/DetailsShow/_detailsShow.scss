.detailShow{
    display: flex;
    justify-content: space-evenly;
    align-items: flex-start;
    align-content: center;
    margin-top: 8rem;
    .detailsCard{
      box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
      transition: 0.3s;
      width: 22vw;
      img {
        border-radius: 41px 41px 41px 41px;
        width: 100%;
      }
      .detailsContainer {
        padding: 2px 16px;
        .pokeName{
          margin: 0%;
        }
        h4{
          margin: 0%;
        }
        .pokeType{
          color: #0d830bfa;
          margin: 0%;
        }
        .pokeWeak{
          color: #e22609fa;
          margin: 0%;
        }
        .pokeHw{
          color: #0914e2fa;
          margin: 0%;
        }
        .pokeNp{
          color: #09e2d0fa;
          margin: 0%;
          margin-bottom: 10px;
        }
      }
    }
  }