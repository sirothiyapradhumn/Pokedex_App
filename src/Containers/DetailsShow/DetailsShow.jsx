import React from 'react'
import { useEffect, useState } from 'react'
import {Link} from 'react-router-dom'
import style from './detailsShow.module.scss'

function DetailsShow({pokeData, apiData}) {
    // console.log(pokeData);
    // console.log(apiData);
    const [nextEvo, setNextEvo] = useState([]);
    const [prevEvo, setPrevEvo] = useState([]);
    const [row, setRow] = useState(pokeData);

    // console.log(row);

    useEffect(()=>{
      pokeData.hasOwnProperty('next_evolution') ? setNextEvo(pokeData.next_evolution) : setNextEvo([]);
      pokeData.hasOwnProperty('prev_evolution') ? setPrevEvo(pokeData.prev_evolution) : setPrevEvo([]);
    },[])

    // console.log(nextEvo);
    // console.log(prevEvo);
    const handleClick = (num)=>{
      var filteRow = apiData.filter((obj)=>obj.num == num);
      // console.log(num, filteRow );
      setRow(filteRow[0]);
      filteRow[0].hasOwnProperty('next_evolution') ? setNextEvo(filteRow[0].next_evolution) : setNextEvo([]);
      filteRow[0].hasOwnProperty('prev_evolution') ? setPrevEvo(filteRow[0].prev_evolution) : setPrevEvo([]);
    }

  return (
    <div className={style.detailShow}>
        <div className={style.detailsCard} >
            <img src={row.img} alt={row.name}/>
            <div className={style.detailsContainer}>
                <h4><b>ID: {row.num}</b></h4>
                <p className={style.pokeName}><b>{row.name}</b></p> 
                <p className={style.pokeType}>Types: {row.type.map((typ)=>typ+"  ")}</p>
                <p className={style.pokeWeak}>Weaknesses: {row.weaknesses.map((weak)=>weak+"  ")}</p>
                <p className={style.pokeHw}>Height: {row.height}</p>
                <p className={style.pokeHw}>Weight: {row.weight}</p>

                <p className={style.pokeNp}>
                Prev Evolution: {row.hasOwnProperty('prev_evolution') ? row.prev_evolution.map(obj=>"> "+obj.name) : "****"}
                </p> 

                <p className={style.pokeNp}>
                  Next Evolution: {row.hasOwnProperty('next_evolution') ? row.next_evolution.map((obj)=>{return "< "+obj.name}) : "****"}
                </p>
            </div>
            <div className={style.btnContainer}>
              {prevEvo.map(obj=>{
                  return <button className={style.btn} variant="outlined" onClick={()=>handleClick(obj.num)}>
                  {obj.name}
                  </button>
                })}
              {nextEvo.map(obj=>{
                return <button className={style.btn} variant="outlined" onClick={()=>handleClick(obj.num)}>
                {obj.name}
                </button>
              })}
            </div>
        </div>

        <button><Link to="/" style={{textDecoration:"none"}}>Back</Link></button>
    </div>
    
  )
}

export default DetailsShow