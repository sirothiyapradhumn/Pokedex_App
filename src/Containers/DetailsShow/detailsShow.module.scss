.detailShow {
  display: flex;
  justify-content: space-evenly;
  align-items: flex-start;
  align-content: center;
  background-color: #1f2833;
  height: 100vh;
  .detailsCard {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    transition: 0.3s;
    width: 22vw;
    background-color: #c5c6c7;
    border-radius: 1rem;
    margin-top: 4rem;
    img {
      border-radius: 41px 41px 41px 41px;
      width: 100%;
    }
    .btnContainer{
      margin: 0.8rem;
    }
    .detailsContainer {
      padding: 2px 16px;
      .pokeName {
        margin: 0%;
      }
      h4 {
        margin: 0%;
      }
      .pokeType {
        color: #0d830bfa;
        margin: 0%;
      }
      .pokeWeak {
        color: #e22609fa;
        margin: 0%;
      }
      .pokeHw {
        color: #0914e2fa;
        margin: 0%;
      }
      .pokeNp {
        color: rgb(216, 113, 10);
        margin: 0%;
      }
    }
  }
}

.btn {
  -webkit-border-radius: 28;
  -moz-border-radius: 28;
  border-radius: 20px;
  font-family: Arial;
  color: #ffffff;
  font-size: 15px;
  background: #0b0c10;
  padding: 10px 20px 10px 20px;
  text-decoration: none;
  margin-left: 10px;
}

.btn:hover {
  background: #414345;
  text-decoration: none;
}