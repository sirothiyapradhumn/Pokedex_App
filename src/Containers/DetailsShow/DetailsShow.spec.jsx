import { shallow } from "enzyme";
import DetailsShow from './DetailsShow'


describe("DetailsShow component", () => {
    let mockrow = {
      id: 1,
      num: "001",
      name: "Bulbasaur",
      img: "http://www.serebii.net/pokemongo/pokemon/001.png",
      type: ["Grass", "Poison"],
      height: "0.71 m",
      weight: "6.9 kg",
      candy: "Bulbasaur Candy",
      candy_count: 25,
      egg: "2 km",
      spawn_chance: 0.69,
      avg_spawns: 69,
      spawn_time: "20:00",
      multipliers: [1.58],
      weaknesses: ["Fire", "Ice", "Flying", "Psychic"],
      next_evolution: [
        {
          num: "002",
          name: "Ivysaur",
        },
        {
          num: "003",
          name: "Venusaur",
        },
      ],
    };

    it("snapshot for ", () => {
        let wrapper = shallow(<DetailsShow pokeData={mockrow} />);
        // console.log(wrapper.debug());
        expect(wrapper).toMatchSnapshot();
    }); 
});