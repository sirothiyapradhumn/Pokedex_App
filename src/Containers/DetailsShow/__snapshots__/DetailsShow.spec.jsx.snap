// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsShow component snapshot for  1`] = `
<div
  className="detailShow"
>
  <div
    className="detailsCard"
  >
    <img
      alt="Bulbasaur"
      src="http://www.serebii.net/pokemongo/pokemon/001.png"
    />
    <div
      className="detailsContainer"
    >
      <h4>
        <b>
          ID: 
          001
        </b>
      </h4>
      <p
        className="pokeName"
      >
        <b>
          Bulbasaur
        </b>
      </p>
      <p
        className="pokeType"
      >
        Types: 
        Grass  
        Poison  
      </p>
      <p
        className="pokeWeak"
      >
        Weaknesses: 
        Fire  
        Ice  
        Flying  
        Psychic  
      </p>
      <p
        className="pokeHw"
      >
        Height: 
        0.71 m
      </p>
      <p
        className="pokeHw"
      >
        Weight: 
        6.9 kg
      </p>
      <p
        className="pokeNp"
      >
        Next/Prev Evolution: 
        Ivysaur
      </p>
    </div>
  </div>
  <button>
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/"
    >
      Back
    </Link>
  </button>
</div>
`;
